LDAP/ldap2s3/ldap2s3.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+import boto3
+from datetime import datetime
+import io
+import json
+from ldap3 import Server, Connection, ALL, SUBTREE
+import lzma
+import logging
+import os
+import yaml
+import sys
+
+
+def setup_logging(stream=sys.stderr, level=logging.INFO):
+    formatstr="[%(asctime)s] %(levelname)s [%(name)s.%(funcName)s:%(lineno)d] %(message)s"
+    logging.basicConfig(format=formatstr, datefmt="%H:%M:%S", stream=stream)
+    logger = logging.getLogger(__name__)
+    logger.setLevel(level)
+    return logger
+
+class DotDict(dict):
+    """
+    dict.item notation for dict()'s
+    """
+    __getattr__ = dict.__getitem__
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def __init__(self, dct):
+        for key, value in dct.items():
+            if hasattr(value, 'keys'):
+                value = DotDict(value)
+            self[key] = value
+
+    def __getstate__(self):
+        return self.__dict__
+
+class ldaper():
```

Comment:
i expect this class to grow if we grab a few more attributes
i had some code around that uses the user profile schema and validation, but we ended up deciding not to use it since it's simply an uploader script and wouldnt do the user profile integration itself